### PR TITLE
Sp comments

### PR DESCRIPTION
**Pull request by ded-possum**
# Description

This is a new feature which users can view a single post's comments by way of the View Comments button link on a given post.  When a single post is being viewed, clicking on the View Comments button will navigate to the comments for that post, displayed in chronological order by date of submission.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Testing Instructions

- run `git add .` then `git commit` all changes in your terminal
- `git checkout main` then `git fetch -a` to get all branches that haven't been fetched yet
- `git checkout SP-comments`
- `start ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid.sln`
- Run these queries in your SQL Object Explorer [Create DB](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/01_Tabloid_Create_DB.sql) and [Seed Data](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/02_Tabloid_Seed_Data.sql)

- In Visual Studio, run in debugger using the `Tabloid` server
- `cd ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid/client && npm start`
- Log in as 'foo@bar.com'
- Click on the Posts link in the nav bar
- Click on the title of the first post you see it should be [grow frictionless e-tailers](http://localhost:3000/posts/16)
- Once there you should see a single post with a View Comments button
- Click the View Comments button to view the single lorem ipsum comment provided by the SQL Query.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works

### CODE REVIEW

File: Tabloid/Repositories/CommentRepository.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                SELECT c.Id, c.UserProfileId, Subject, c.Content, c.CreateDateTime, u.FirstName, u.LastName, u.DisplayName, 
+                    u.Email, 
+                    u.CreateDateTime AS UserProfileDateCreated,
+                    u.ImageLocation AS AvatarImage,
+                    u.UserTypeId 
+                  FROM Comment c
+                  JOIN Post p on c.PostId = p.Id
+                  Join UserProfile u on p.UserProfileId = u.Id
+              WHERE PostId = @Id";
+
```
Comment (andy-birt) [inline Tabloid/Repositories/CommentRepository.cs:33]: Here you will want to add an order by statement. According to the ticket, the comments should be ordered by creation date with the most recent at the top.

File: Tabloid/Repositories/CommentRepository.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                            {
+                                Id = DbUtils.GetInt(reader, "UserProfileId"),
+                                FirstName = DbUtils.GetString(reader, "FirstName"),
+                                LastName = DbUtils.GetString(reader, "LastName"),
+                                DisplayName = DbUtils.GetString(reader, "DisplayName"),
+                                Email = DbUtils.GetString(reader, "Email"),
+                                CreateDateTime = DbUtils.GetDateTime(reader, "UserProfileDateCreated"),
+                                ImageLocation = DbUtils.GetString(reader, "AvatarImage"),
+                                UserTypeId = DbUtils.GetInt(reader, "UserTypeId")
+                            }
```
Comment (andy-birt) [inline Tabloid/Repositories/CommentRepository.cs:60]: You could also make a post object here, you're already joining the post from the comment. You could add the title of the post in the select statement. You might have to also select other parts of the post as well but in this case we only really need the title.

File: Tabloid/client/src/components/comment/PostComment.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+useEffect(() => {
+    getCommentsByPostId(id);
+}, []);
+
+return (
+    <>
+    {comments.length > 0 ? 
+        <>
+        <Container className="justify-content-center" fluid={true}>
+                {comments.map((comment, index) => {
```
Comment (andy-birt) [inline Tabloid/client/src/components/comment/PostComment.js:20]: Here you can add the title of the post before the comments map begins and add a button that redirects back to the post details page

File: Tabloid/client/src/components/comment/PostComment.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext, useEffect } from "react";
+import { CommentContext } from "../../providers/CommentProvider";
+import { Container, Table } from "reactstrap";
+import Comment from "./Comment";
+import { Link } from "react-router-dom";
+import { useHistory, useParams } from "react-router-dom";
+
+const CommentList = () => {
```
Comment (andy-birt) [inline Tabloid/client/src/components/comment/PostComment.js:8]: If you don't want to get all the post stuff from the comment query you could also possibly get the current post from the post context.

**Review by andy-birt — CHANGES_REQUESTED** [review]
Steven, I love your PR. Great instruction! There are a few things, however, we need to consider. Just looking back at the ticket, I left some comments for you to take a look at regarding the ticket requirements and hope you find them helpful.

File: Tabloid/Repositories/CommentRepository.cs (hunk 1 above)
Comment (ded-possum) [reply Tabloid/Repositories/CommentRepository.cs:33]: Checked off!

File: Tabloid/Repositories/CommentRepository.cs (hunk 2 above)
Comment (ded-possum) [reply Tabloid/Repositories/CommentRepository.cs:60]: Went with js suggestion below in lieu of

File: Tabloid/client/src/components/comment/PostComment.js (hunk 3 above)
Comment (ded-possum) [reply Tabloid/client/src/components/comment/PostComment.js:20]: Done!

File: Tabloid/client/src/components/comment/PostComment.js (hunk 4 above)
Comment (ded-possum) [reply Tabloid/client/src/components/comment/PostComment.js:8]: Done!

**Review by andy-birt — APPROVED** [review]
:burrito: sweeto burrito

**Review by helloamandaball — APPROVED** [review]
🌮 🌮 🌮 I can view comments for each post. Good job!

**Pull request merged**